domain/snyk/scanner/scanner.go

```diff
@@ -263,27 +266,26 @@ func (sc *DelegatingConcurrentScanner) Scan(
 
 	sc.scanNotifier.SendInProgress(folderPath)
 	gitCheckoutHandler := vcs.NewCheckoutHandler()
-	defer func() {
-		if gitCheckoutHandler.CleanupFunc() != nil {
-			logger.Debug().Msg("Calling cleanup func for base folder")
-			gitCheckoutHandler.CleanupFunc()()
-		}
-	}()
+
 	waitGroup := &sync.WaitGroup{}
+	referenceBranchScanWaitGroup := &sync.WaitGroup{}
 	for _, scanner := range sc.scanners {
 		if scanner.IsEnabled() {
 			waitGroup.Add(1)
+			referenceBranchScanWaitGroup.Add(1)
 			go func(s snyk.ProductScanner) {
 				defer waitGroup.Done()
 				span := sc.instrumentor.NewTransaction(context.WithValue(ctx, s.Product(), s), string(s.Product()), method)
 				defer sc.instrumentor.Finish(span)
 				logger.Info().Msgf("Scanning %s with %T: STARTED", path, s)
 				// TODO change interface of scan to pass a func (processResults), which would enable products to stream
+				sc.scanStateAggregator.SetScanInProgress(folderPath, scanner.Product(), false)
```

Comment:
The positioning of the comment is now off. Should be around `sc.internalScan` I guess.